File: Server/GridiaServer/src/main/java/hoten/gridia/content/ItemUse.java
package hoten.gridia.content;

import java.util.ArrayList;
import java.util.List;
import org.apache.commons.lang.builder.ReflectionToStringBuilder;

public class ItemUse {

    public String name, successMessage, failureMessage, focusSubType, animation;
    public int tool, focus, focusQuantityConsumed = 1, toolQuantityConsumed = 1, successTool = -1;
    public int surfaceGround = -1;
    public List<Integer> products = new ArrayList();
    public List<Integer> quantities = new ArrayList();

    @Override
    public String toString() {
        return ReflectionToStringBuilder.toString(this);
    }
}

File: client/Assets/GridiaClient/Content/ItemUse.cs
using System.Collections.Generic;

namespace Gridia
{
    public class ItemUse
    {
        public int Tool;
        public int Focus;
        public int SuccessTool;
        public List<int> Products;

        public int GetIdentifyingItem()
        {
            if (SuccessTool > 0)
            {
                return SuccessTool;
            }
            return Products.Count != 0 ? Products[0] : 1;
        }
    }
}
